compiler/rustc_span/src/symbol.rs

```diff
@@ -1966,51 +1966,51 @@ pub(crate) struct Interner(Lock<InternerInner>);
 #[derive(Default)]
 struct InternerInner {
     arena: DroplessArena,
-    names: FxHashMap<&'static str, Symbol>,
-    strings: Vec<&'static str>,
+    names: FxHashMap<*const str, Symbol>,
```

Comment:
This `Hash` impl hashes the pointer address instead of the string itself. You need to write a newtype wrapper that actually hashes the string.

Reply:
> As long as the look-up occurs via this address too...

Every string you'd be looking up would possibly have a different address, even if they're the same string, which kinda defeats the purpose of interning at all.

Reply:
Ahhh 👍